src/utils/content.ts

```diff
@@ -0,0 +1,143 @@
+import {nip19} from "nostr-tools";
+
+export enum ContentType {
+  text = "text",
+  newline = 'newline',
+  mention = 'mention',
+  topic = 'topic',
+  link = 'link'
+}
+
+export type Content = {
+  type: ContentType;
+  value: string;
+}
+
+export enum MediaType {
+  image = "image",
+  video = "video",
+  link = "link"
+}
+
+export type Media = {
+  type: MediaType;
+  url: string;
+}
+
+export const parseContent = content => {
+  const text = content.trim();
+  const result: object[] = [];
+  let buffer = "", i = 0;
+
+  const push = (type, text, value = null) => {
+    if (buffer) {
+      result.push({type: ContentType.text, value: buffer});
+      buffer = "";
+    }
+
+    result.push({type, value: value || text});
+    i += text.length;
+  }
+
+  for (; i < text.length; ) {
+    const tail = text.slice(i);
+
+    const newLine = tail.match(/^\n+/);
+
+    if (newLine) {
+      push(ContentType.newline, newLine[0]);
+      continue;
+    }
+
+    const mentionMatch = tail.match(/^#\[\d+\]/i);
+
+    if (mentionMatch) {
+      push(ContentType.mention, mentionMatch[0]);
+      continue;
+    }
+
+    const topicMatch = tail.match(/^#\w+/i);
+
+    if (topicMatch) {
+      push(ContentType.topic, topicMatch[0]);
+      continue;
+    }
+
+    const bech32Match = tail.match(/^(nostr:)?n(event|ote|profile|pub)1[\d\w]+/i);
+
+    if (bech32Match) {
+      try {
+        const entity = bech32Match[0].replace("nostr:", "");
+        const {type, data} = nip19.decode(entity) as {type: string; data: object};
```

Comment:
can you remove the `nostr-tool` and use https://github.com/digi-monkey/flycat-web/blob/master/src/service/api.ts#L831 to do `nip19.decode(entity)` instead?

Reply:
yes